<!--

Programmer: Victoria Albanese
Filename: ufo.launch

Purpose: Launch multiple Intel RealSense 435i sensors
so that the pointclouds they generate can be visualized

-->

<launch>

    <!-- This node manages all of the incoming passthrough filter nodelets -->
    <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen" />


    <!-- This node rotates all the cameras into a global frame -->    
    <node pkg="synchronize_pointclouds" type="transformer_node" name="transformer_node" />


    <!-- This launches each of the cameras with parameters given in separate launchfiles -->
    <include file="$(find synchronize_pointclouds)/launch/camera_1.launch" />
    <include file="$(find synchronize_pointclouds)/launch/camera_2.launch" />
    <include file="$(find synchronize_pointclouds)/launch/camera_3.launch" />
    <include file="$(find synchronize_pointclouds)/launch/camera_4.launch" />
    <include file="$(find synchronize_pointclouds)/launch/camera_5.launch" />
    <include file="$(find synchronize_pointclouds)/launch/camera_6.launch" />
    <include file="$(find synchronize_pointclouds)/launch/camera_7.launch" />
    <include file="$(find synchronize_pointclouds)/launch/camera_8.launch" />
  
    <!-- This launches RViz all setup to visualize them -->
    <!--
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find synchronize_pointclouds)/rviz/spidr.rviz" required="true" />
    -->

</launch>

